napi-inl.h

```diff
@@ -40,6 +43,14 @@ namespace details {
     return __VA_ARGS__;                                  \
   }
 
+// Usually ony could use NAPI_THROW_IF_FAILED(env, status, void()),
```

Comment:
How about we re-phrase this to
```C++
// We need a _VOID version of this macro to avoid warnings resulting from
// leaving the NAPI_THROW_IF_FAILED `...` argument empty..
```